### PR TITLE
docs: restructure the concepts page

### PR DESCRIPTION
**Pull request by jannyHou**
connect to https://github.com/strongloop/loopback-next/issues/5776

I renamed the concept items to use the singular terms.

fold subsections:
Screen Shot 2020-08-10 at 3 31 45 PM

unfold subsections, part 1:
Screen Shot 2020-08-10 at 3 32 10 PM

unfold subsections, part 2:
Screen Shot 2020-08-10 at 3 32 19 PM

## Checklist

👉 [Read and sign the CLA (Contributor License Agreement)](https://cla.strongloop.com/agreements/strongloop/loopback-next) 👈

- [ ] `npm test` passes on your machine
- [ ] New tests added or existing tests modified to cover all changes
- [ ] Code conforms with the [style guide](http://loopback.io/doc/en/contrib/style-guide.html)
- [ ] API Documentation in code was updated
- [ ] Documentation in [/docs/site](../tree/master/docs/site) was updated
- [ ] Affected artifact templates in `packages/cli` were updated
- [ ] Affected example projects in `examples/*` were updated

👉 [Check out how to submit a PR](https://loopback.io/doc/en/lb4/submitting_a_pr.html) 👈

### CODE REVIEW

File: docs/site/Interceptors.md [hunk 1]
```diff
@@ -1,9 +1,9 @@
 ---
 lang: en
-title: 'Interceptors'
+title: 'Interceptor'
 keywords: LoopBack 4.0, LoopBack 4, Node.js, TypeScript, OpenAPI, Interceptor
 sidebar: lb4_sidebar
-permalink: /doc/en/lb4/Interceptors.html
+permalink: /doc/en/lb4/Interceptor.html
```
Comment (dhmlau) [inline docs/site/Interceptors.md:6]: I think I understand the rationale behind changing the html path name is to make it consistent with the title. If that's the case, can we keep the same html and add the `redirect_to`? Renaming the html path will cause broken links in other docs or articles (mostly external ones, not part of LB docs). WDYT?

Comment (jannyHou) [reply docs/site/Interceptors.md:6]: @dhmlau 👍 Fair enough, adding.

**Comment by raymondfeng** [comment]
I start to lean toward getting rid of `TypeScript` and move `Decorators` and `Mixins` to other sections.

- Decorators are used to supply metadata (it seems weird to list specific decorators under TypeScript)
- Mixins are a way to compose LoopBack 4 classes

**Comment by jannyHou** [comment]
Thanks @raymondfeng fair enough, I moved them into "Core", see the updated doc:
Screen Shot 2020-08-11 at 6 37 09 PM

**Comment by agnes512** [comment]
Let's update https://loopback.io/doc/en/lb4/index.html#how-is-our-documentation-organized

**Comment by agnes512** [comment]
Are we changing the order in this PR?

**Comment by jannyHou** [comment]
@agnes512 Thanks for the suggestion in https://github.com/strongloop/loopback-next/pull/6085#issuecomment-672938916 good catch, and yes the order is changed as well.

File: docs/site/Component.md [hunk 2]
```diff
@@ -1,9 +1,10 @@
 ---
 lang: en
-title: 'Components'
+title: 'Component'
 keywords: LoopBack 4.0, LoopBack 4, Node.js, TypeScript, OpenAPI, Concepts
 sidebar: lb4_sidebar
-permalink: /doc/en/lb4/Components.html
+permalink: /doc/en/lb4/Component.html
+redirect_from: /doc/en/lb4/Components.html
```
Comment (bajtos) [inline docs/site/Component.md:7]: +1 to use singular form in `permalink` and add `redirect_from` to preserve existing links 👍🏻

**Pull request merged**